compiler/luci/service/src/Nodes/CirclePad.test.cpp

```diff
@@ -31,3 +33,42 @@ TEST(CloneNodeTest, clone_Pad)
   auto cloned_pad = dynamic_cast<luci::CirclePad *>(cloned);
   ASSERT_NE(nullptr, cloned_pad);
 }
+
+TEST(ShapeRuleTest, pad_dynamic_shape)
```

Comment:
plz add negative test with `_NEG` suffix having count(positive) <= count(negative)

Reply:
Just to be clear, do you mean testing for pad inference if it fails?
I don't understand 
> suffix having count(positive) <= count(negative)
this line.

Reply:
> Just to be clear, do you mean testing for pad inference if it fails?
 
Yes, you're right. 

---------------------------------------

( I put my understanding here ) 

Negative tests mean kinds of these 
  * When unexpected data are given, it throws an error as expected (make sure that the system does not just die)
  * ... :thinking: 

As I know, there is a policy that we should have more negative tests than positive tests. 

You can refer this PR (https://github.com/Samsung/ONE/pull/13698). There are two negative tests and one positive test. Also, negative tests have suffix `_NEG` to mark that it is negative test.

Reply:
> You can refer this PR (#13698). There are two negative tests and one positive test. Also, negative tests have suffix `_NEG` to mark that it is negative test.

Thanks for kind reply. I'll add some negative tests. 
In my opinion, there are some negative cases with pad operations.

&emsp; 1.  paddings shape is not [n,2]
&emsp; 2.   paddings is over 32 bit limit

Is there any other suggestions?